rust/opendp/src/trans/count.rs

```diff
@@ -28,6 +28,24 @@ pub fn make_count<MI, MO, TI>() -> Fallible<Transformation<VectorDomain<AllDomai
         StabilityRelation::new_from_constant(MO::Distance::one())))
 }
 
+
+pub fn make_count_distinct<MI, MO, TI>() -> Fallible<Transformation<VectorDomain<AllDomain<TI>>, AllDomain<MO::Distance>, MI, MO>>
+    where MI: DatasetMetric,
+          MO: SensitivityMetric,
+          MO::Distance: TryFrom<usize> + Bounded + One + DistanceConstant,
+          TI: Eq + Hash {
+    Ok(Transformation::new(
+        VectorDomain::new_all(),
+        AllDomain::new(),
+        Function::new(move |arg: &Vec<TI>| {
+            let len = arg.iter().collect::<HashSet<_>>().len();
+            MO::Distance::try_from(len).unwrap_or(MO::Distance::max_value())
```

Comment:
Rather than return `max_value()`, wouldn't it be better to fail?

Reply:
I could set TO=i8, and if the function fails, I know the data has at least 128 elements. Then repeat for each datatype to get a rough bound on the power-of-2 length of the private data, without being charged for this information.

Reply:
Good point!